structuretoolkit/build/mesh.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+import warnings
+from ase.atoms import Atoms
+
+
+def create_mesh(
+    structure: Atoms,
+    n_mesh: int|list = 10,
+    density: float|None = None,
+    endpoint: bool = False
+):
+    """
+    Create a mesh based on a structure
+
+    Args:
+        structure (ase.atoms.Atoms): ASE Atoms
+        n_mesh (int): Number of grid points in each direction. If one number
+            is given, it will be repeated in every direction (i.e. n_mesh = 3
+            is the same as n_mesh = [3, 3, 3])
+        density (float): Density of grid points. Ignored when n_mesh is not
+            None
+        endpoint (bool): Whether both the edges get separate points or not.
+            cf. endpoint in numpy.linspace
+
+    Returns:
+        (n, n, n, 3)-array: mesh
```

Comment:
Does it though? What happens when I invoke `create_mesh(some_structure, n_mesh=(4, 4, 4, 4))`? I'm not sure how the einsum will handle it, but up until that point the rest of the code looks like it will happily give you a hyper mesh. Same problem with `n_mesh=(4,4)`, although there I'm more confident the einsum will fail because it over-specifies indices (although it's not obvious to me that the error message will be clear and helpful).

If you want this to be restricted to three-dimensional data, some checking of `n_mesh` will be required.

Reply:
Now it checks the shape